packages/dd-trace/src/appsec/callbacks/ddwaf.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const log = require('../../log')
+const Addresses = require('../addresses')
+const Gateway = require('../../gateway/engine')
+const Reporter = require('../reporter')
+
+let warned = false
+
+const validAddressSet = new Set(Object.values(Addresses))
+
+const DEFAULT_MAX_BUDGET = 5e3 // µs
+
+// TODO: put reusable code in a base class
+class WAFCallback {
+  static loadDDWAF (rules) {
+    try {
+      // require in `try/catch` because this can throw at require time
+      const { DDWAF } = require('@datadog/native-appsec')
+
+      return new DDWAF(rules)
+    } catch (err) {
+      if (!warned) {
```

Comment:
We used to beta test security rules and other stuff, by running them on a different instance of the WAF, that's how it can be ran multiple times. Also disabling then re-enabling appsec will do that too.

I do not understand what you mean by "a reference global to the file".

Reply:
Removed the check, it's not required and will be superseeded by the logs PR anyway